cluster-autoscaler/simulator/predicates.go

```diff
@@ -103,85 +69,58 @@ func (NoOpEventRecorder) AnnotatedEventf(object runtime.Object, annotations map[
 
 // NewPredicateChecker builds PredicateChecker.
 func NewPredicateChecker(kubeClient kube_client.Interface, stop <-chan struct{}) (*PredicateChecker, error) {
-	staticInitIfNeeded()
-
 	informerFactory := informers.NewSharedInformerFactory(kubeClient, 0)
-	podInformer := informerFactory.Core().V1().Pods()
+	providerRegistry := algorithmprovider.NewRegistry(1)
+	config := providerRegistry[scheduler_apis_config.SchedulerDefaultProviderName]
+	snapshot := NewEmptySnapshot()
 
-	sched, err := scheduler.New(
+	volumeBinder := scheduler_volumebinder.NewVolumeBinder(
 		kubeClient,
-		informerFactory,
-		podInformer,
-		NoOpEventRecorder{},
-		stop,
-		scheduler.WithFrameworkConfigProducerRegistry(nil),
+		informerFactory.Core().V1().Nodes(),
+		informerFactory.Storage().V1().CSINodes(),
+		informerFactory.Core().V1().PersistentVolumeClaims(),
+		informerFactory.Core().V1().PersistentVolumes(),
+		informerFactory.Storage().V1().StorageClasses(),
+		time.Duration(10)*time.Second,
 	)
-	if err != nil {
-		return nil, fmt.Errorf("couldn't create scheduler; %v", err)
-	}
-
-	predicateMap := map[string]predicates.FitPredicate{}
-	for predicateName, predicateFunc := range sched.Algorithm.Predicates() {
-		predicateMap[predicateName] = predicateFunc
-	}
-	// We want to make sure that some predicates are present to run them first
-	// as they are cheap to check and they should be enough to fail predicates
-	// in most of our simulations (especially binpacking).
-	if _, found := predicateMap["PodFitsResources"]; !found {
-		predicateMap["PodFitsResources"] = predicates.PodFitsResources
-	}
-	if _, found := predicateMap["PodToleratesNodeTaints"]; !found {
-		predicateMap["PodToleratesNodeTaints"] = predicates.PodToleratesNodeTaints
-	}
 
-	predicateList := make([]PredicateInfo, 0, len(predicateMap))
-	for _, predicateName := range priorityPredicates {
-		if predicate, found := predicateMap[predicateName]; found {
-			predicateList = append(predicateList, PredicateInfo{Name: predicateName, Predicate: predicate})
-			delete(predicateMap, predicateName)
-		}
-	}
+	framework, err := scheduler_framework.NewFramework(
+		scheduler_plugins.NewInTreeRegistry(&scheduler_plugins.RegistryArgs{
+			VolumeBinder: volumeBinder,
+		}),
+		config.FrameworkPlugins,
+		config.FrameworkPluginConfig,
+		scheduler_framework.WithClientSet(kubeClient),
```

Comment:
Do we need to pass kubeClient if we're passing informers and we very much don't want to do any mutating calls?

Reply:
supposedly not needed. Dropped. Will verify in e2e tests if it actually works.